<launch>
  <machine name="local" address="localhost" default="true" />

  <param name="use_sim_time" value="true"/>
  <arg name="rate" default="1.0" />

  <node name="tinySLAM" pkg="slam_constructor" type="tiny_slam"
        cwd="node" required="true" output="screen">

    <!-- = Mapping = -->
    <param name="slam/mapping/blur" value="0.5" />
    <!-- * Occupancy Estimator setup -->
    <param name="slam/occupancy_estimator/type" value="const" />
    <param name="slam/occupancy_estimator/base_occupied/prob" value="0.95" />
    <param name="slam/occupancy_estimator/base_empty/prob" value="0.01" />
    <!-- * Occupancy Map setup -->
    <param name="slam/map/height_in_meters" value="100" />
    <param name="slam/map/width_in_meters" value="100" />
    <param name="slam/map/meters_per_cell" value="0.1" />
    <!-- * FIXME -->
    <param name="slam/cell/type" value="avg" />

    <!-- = Localization = -->
    <param name="slam/scmtch/type" value="MC" />
    <param name="slam/scmtch/MC/dispersion/translation" value="0.2" />
    <param name="slam/scmtch/MC/dispersion/rotation" value="0.1" />
    <param name="slam/scmtch/MC/dispersion/failed_attempts_limit" value="20" />
    <param name="slam/scmtch/MC/attempts_limit" value="100" />
    <!-- <param name="slam/scmtch/MC/seed" value="666666" /> -->

    <param name="slam/scmtch/spe/type" value="wmpp" />
    <param name="slam/scmtch/spe/wmpp/weighting/type" value="even" />

    <!-- = Input = -->
    <param name="in/lscan2D/ros/topic/name" value="/base_scan" />
    <param name="in/odometry/ros/tf/odom_frame_id" value="odom_combined" />

    <!-- Internal constants for ros-->
    <param name="ros/skip_exceeding_lsr_vals" value="false" />
    <param name="ros/rviz/map_publishing_rate" value="5.0" />
    <param name="ros/tf/buffer_duration" value="5.0" /> <!-- in seconds-->
    <param name="ros/filter_queue_size" value="1000" />
    <param name="ros/subscribers_queue_size" value="1000" />
  </node>

  <node name="player" pkg="rosbag" type="play"
        args="--delay=3 --clock --rate $(arg rate) $(arg path)" />

  <node name="rviz" pkg="rviz" type="rviz" respawn="true"
        args="-d $(find slam_constructor)/rviz/debug.rviz" />

  <!-- Uncomment this node if a robot trajectory needs to be published
    <node name="path_publisher" pkg="slam_constructor" type="path_publisher">
      <param name="wait_for_transform_duration" value="3.0" />
      <param name="base_link_frame_id" value="base_link" />
      <param name="subscribers_queue_size" value="1000" />
      <param name="publishing_rate" value="5.0" />
      <param name="path_topic_name" value="/path" />
    </node>
  -->
</launch>
